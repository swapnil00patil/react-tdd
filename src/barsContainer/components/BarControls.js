import React from 'react'
import styled from 'styled-components'

const Button = styled.button`
  text-align:center;
  cursor: pointer;
  padding: 5px 10px;
  border-radius: 5px;
  font-size: 18px;
  margin-right: 8px;
  min-width: 50px;

  &:hover {
    background: gray;
    color:#fff;
  }
`
const Placeholder = styled.div`
  text-align:center;
  cursor: pointer;
  padding: 5px 10px;
  border-radius: 5px;
  font-size: 18px;
  margin-right: 8px;
  padding: 5px 20px;
  border: 1px solid #ccc;
  display: inline;

  &:hover {
    background: gray;
    color:#fff;
  }
`

class BarControls extends React.PureComponent {
  render () {
    const { buttons, updateProgress } = this.props
    return (buttons && buttons.length > 0) ? buttons.map((value, index) => [
      <Button key={index} onClick={() => updateProgress(value)}>{value}</Button>
    ]) : [1, 2, 3, 4].map((value, index) => <Placeholder key={index} />)
  }
}

export default BarControls
