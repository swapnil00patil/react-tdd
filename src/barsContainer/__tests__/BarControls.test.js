import React from 'react'
import ReactDOM from 'react-dom'
import { shallow, mount } from 'enzyme'

import BarControls from '../components/BarControls'

describe('BarControls', () => {
  it('BarControls should be defined', () => {
    expect(BarControls).toBeDefined()
  })

  it('should render exactly same number bars', () => {
    const wrapper = mount(<BarControls buttons={[12, 13, 14]} />)
    wrapper.find('button').length.valueOf(3)
  })

  it('check correct value is retrieved when clicked on button', () => {
    const value = 12
    const onClick = jest.fn()
    const wrapper = mount(<BarControls buttons={[12, 13, 14]} updateProgress={onClick} />)

    wrapper.first('button').simulate('click', {
      target: { value }
    })

    expect(onClick).toBeCalledWith(value)
  })

  it('should match snapshot', () => {
    const wrapper = shallow(<BarControls buttons={[12, 13, 14]} />)
    expect(wrapper).toMatchSnapshot()
  })
})
