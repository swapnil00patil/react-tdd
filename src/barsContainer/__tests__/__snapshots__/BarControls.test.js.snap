// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarControls should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BarControls
    buttons={
        Array [
            12,
            13,
            14,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    Array [
      <styled.button
        onClick={[Function]}
>
        12
</styled.button>,
    ],
    Array [
      <styled.button
        onClick={[Function]}
>
        13
</styled.button>,
    ],
    Array [
      <styled.button
        onClick={[Function]}
>
        14
</styled.button>,
    ],
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      Array [
        <styled.button
          onClick={[Function]}
>
          12
</styled.button>,
      ],
      Array [
        <styled.button
          onClick={[Function]}
>
          13
</styled.button>,
      ],
      Array [
        <styled.button
          onClick={[Function]}
>
          14
</styled.button>,
      ],
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
